contracts/utils/Bytes.sol

```diff
@@ -187,6 +200,32 @@ library Bytes {
         return (value >> 8) | (value << 8);
     }
 
+    /**
+     * @dev Counts the number of leading zeros in a bytes array. Returns `buffer.length`
+     * if the buffer is all zeros.
+     */
+    function clz(bytes memory buffer) internal pure returns (uint256) {
+        for (uint256 i = 0; i < buffer.length; i += 32) {
+            uint256 value = uint256(_unsafeReadBytesOffset(buffer, i));
+            unchecked {
+                // Mask out bytes beyond buffer length
+                // left is buffer.length at most, can't overflow at realistic size
+                uint256 left = buffer.length - i;
+                if (left < 32) {
+                    // left is less than 32, can't overflow
+                    uint256 shift = (32 - left) * 8;
+                    value = (value >> shift) << shift; // Clear the lower bits for the last iteration
+                }
+            }
```

Comment:
Is that part necessary ? If we read beyound the buffer limit there are 2 possible cases:
- the value is 0, we don't get into the `if (value !=0)`, loop doesn't go any further, we get the default value.
- the value is 0, we get into the `if(value != 0)`, and the `Math.min` makes sure we don't count zeros after the buffer.

Reply:
```suggestion
```